packages/@react-spectrum/s2/src/ComboBox.tsx

```diff
@@ -383,11 +431,331 @@ export function ComboBoxSection<T extends object>(props: ComboBoxSectionProps<T>
+          <InputContext.Consumer>
+            {ctx => (
+              <InputContext.Provider value={{...ctx, ref: mergeRefs((ctx as any)?.ref, inputRef)}}>
+                <Input aria-describedby={spinnerId} />
+              </InputContext.Provider>
+            )}
+          </InputContext.Consumer>
+          {isInvalid && <FieldErrorIcon isDisabled={isDisabled} />}
+          {showFieldSpinner && (
+            <ProgressCircle
+              id={spinnerId}
+              isIndeterminate
+              size="S"
+              styles={progressCircleStyles({size, isInput: true})}
+              aria-label={stringFormatter.format('table.loading')} />
+          )}
+          <Button
+            ref={buttonRef}
+            // Prevent press scale from sticking while ComboBox is open.
+            // @ts-ignore
+            isPressed={false}
+            style={renderProps => pressScale(buttonRef)(renderProps)}
+            className={renderProps => inputButton({
+              ...renderProps,
+              size,
+              isOpen
+            })}>
+            <ChevronIcon
+              size={size}
+              className={iconStyles} />
+          </Button>
+        </FieldGroup>
+        <HelpText
+          size={size}
+          isDisabled={isDisabled}
+          isInvalid={isInvalid}
+          description={descriptionMessage}>
+          {errorMessage}
+        </HelpText>
+        <PopoverBase
+          hideArrow
+          triggerRef={triggerRef}
+          offset={menuOffset}
+          placement={`${direction} ${align}` as Placement}
+          shouldFlip={shouldFlip}
+          UNSAFE_style={{
+            width: menuWidth ? `${menuWidth}px` : undefined,
+            // manually subtract border as we can't set Popover to border-box, it causes the contents to spill out
+            '--trigger-width': `calc(${triggerWidth} - 2px)`
+          } as CSSProperties}
+          styles={style({
+            minWidth: '[var(--trigger-width)]',
+            width: '[var(--trigger-width)]'
+          })}>
+          <Provider
+            values={[
+              [HeaderContext, {styles: listboxHeader({size})}],
+              [HeadingContext, {styles: listboxHeading}],
+              [TextContext, {
+                slots: {
+                  'description': {styles: description({size})}
+                }
+              }]
+            ]}>
+            <Virtualizer
+              layout={ListLayout}
+              layoutOptions={{
+                estimatedRowHeight: 32,
+                // TODO: can we get rid of this and instead handle padding else where other than the layout options? Kinda gross that we need to do this check.
+                // Perhaps can consider only applying padding if the collection actually has content since the renderEmptyState content is outside the virtualizer
+                // Otherwise could consider moving renderEmptyState into the collection...
+                padding: isEmptyOrLoading ? 0 : 8,
```

Comment:
we actually don't want any extra padding (or have the virtualizer div render anything really) so that the "No results"/isLoading spinner dictate the height of the the popover. This check is a bit gross but not really sure there is a good way to do this otherwise.